tools/mmp/resolver.cs

```diff
@@ -69,7 +69,7 @@ public AssemblyDefinition AddAssembly (string fileName)
 			if (cache.TryGetValue (name, out assembly))
 				return assembly;
 
-			assembly = AssemblyDefinition.ReadAssembly (fileName, new ReaderParameters { AssemblyResolver = this });
+			assembly = AssemblyDefinition.ReadAssembly (fileName, new ReaderParameters { AssemblyResolver = this, InMemory = true });
```

Comment:
mtouch has a size check and only loads everything into memory if the assembly is < 100 MB:

https://github.com/xamarin/xamarin-macios/blob/master/tools/mtouch/AssemblyResolver.cs#L58

maybe mmp should do the same?

I haven't seen such big assemblies in XM yet, but since it's certainly possible, someone will do it sooner or later.